src/fastoad/model_base/openmdao/group.py

```diff
@@ -0,0 +1,175 @@
+            types=(bool, str),
+            default=self.default_linear_solver,
+            desc="If a string is given like `om.LinearBlockGS`, it will be used "
+            "(convention `import openmdao.api as om' is assumed`. "
+            "If `False` is given, no linear solver is added (i.e. om.LinearRunOnce is used). "
+            'Ignored if option "use_inner_solver" is False',
+            check_valid=_forbid_true_value,
+        )
+        self.options.declare(
+            "nonlinear_solver",
+            types=(bool, str),
+            default=self.default_nonlinear_solver,
+            desc="If a string is given like `om.NewtonSolver`, it will be used "
+            "(convention `import openmdao.api as om' is assumed`. "
+            "If `False` is given, no non-linear solver is added (i.e. om.NonlinearRunOnce is used)."
+            'Ignored if option "use_inner_solver" is False',
+            check_valid=_forbid_true_value,
+        )
+        self.options.declare(
+            "linear_options",
+            types=dict,
+            default=self.default_solver_options["linear_options"],
+            desc="Options for linear solver. This dict will be updated with provided dict, "
+            "meaning that options that are not in provided dict will keep the default value. "
+            'Ignored if "use_inner_solver" is False.',
+        )
+        self.options.declare(
+            "nonlinear_options",
+            types=dict,
+            default=self.default_solver_options["nonlinear_options"],
+            desc="Options for non-linear solver. This dict will be updated with provided dict, "
+            "meaning that options that are not in provided dict will keep the default value. "
+            'Ignored if "use_inner_solver" is False.',
+        )
+
+    def setup(self):
+        super().setup()
+
+        if self.options["use_solvers"]:
+            linear_solver_class = self._get_solver_class("linear_solver")
+            nonlinear_solver_class = self._get_solver_class("nonlinear_solver")
+
+            linear_options = self._get_solver_options("linear_options")
+            nonlinear_options = self._get_solver_options("nonlinear_options")
+
+            if linear_solver_class:
+                self.linear_solver = linear_solver_class(**linear_options)
+            if nonlinear_solver_class:
+                self.nonlinear_solver = nonlinear_solver_class(**nonlinear_options)
+
+    def _get_solver_class(self, option_name: str):
+        """
+        For option_name in ["linear_solver", "nonlinear_solver"], provide the solver class
+        from the string self.options[option_name].
+        Class will be om.<class_name>, whether the stringbegins with "om." or not.
+        """
+        solver_name = self.options[option_name]
+        if isinstance(solver_name, str):
+            if solver_name.startswith("om."):
+                solver_name = solver_name[3:]
+            solver_class = om.__dict__[solver_name]
```

Comment:
Is the solver class something specific from FAST-OAD or does this come from OpenMDAO ?

Reply:
OpenMDAO classes, since they are retrieved from `om` (= `openmdao.api`)